File: backend/EduCore/EduCore.Business/Entities/Seccion.cs
using EduCore.Business.Enums;

namespace EduCore.Business.Entities
{
    public class Seccion
    {
        public int Id { get; private set; }
        public string Nombre { get; private set; }
        public Turno Turno { get; private set; }
        public int GradoId { get; private set; }
        public Grado Grado { get; private set; }
        public int PeriodoId { get; private set; }
        public Periodo Periodo { get; private set; }
        public bool Activo { get; private set; }

        private Seccion() { } // Constructor para EF Core

        public Seccion(string nombre, Turno turno, int gradoId, int periodoId)
        {
            if (string.IsNullOrWhiteSpace(nombre))
                throw new ArgumentException("El nombre de la sección es obligatorio");

            Nombre = nombre;
            Turno = turno;
            GradoId = gradoId;
            PeriodoId = periodoId;
            Activo = true;
        }

        public void Desactivar()
        {
            Activo = false;
        }
    }
}

File: frontend/educore-front/src/app/core/services/seccion.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface SeccionDto {
  id: number;
  nombre: string;
  turno: string;
  numeroGrado: number;
  nivelGrado: string;
  anioPeriodo: number;
  activo: boolean;
}

export interface CrearSeccionDto {
  nombre: string;
  turno: number;
  gradoId: number;
  periodoId: number;
}

@Injectable({
  providedIn: 'root'
})
export class SeccionService {
  readonly apiUrl = `${environment.apiUrl}/Seccion`;

  constructor(private http: HttpClient) {}

  crear(dto: CrearSeccionDto) {
    return this.http.post(this.apiUrl, dto);
  }
}